contracts/src/Gateway.sol

```diff
@@ -428,6 +433,10 @@ contract Gateway is IGateway, IInitializable {
         uint128 destinationFee,
         uint128 amount
     ) external payable {
+        if (destinationFee > MAX_DESTINATION_TRANSFER_FEE) {
```

Comment:
The check has to be done here as immutable variables cannot be shared with libraries without using parameters to pass them down.